### ISSUE TITLE
Selector doesn't get recomputed after the nested record is changed (after bumping version from v0.12.4 to v0.13.0+)

### ISSUE THREAD

**Issue by meltyshev** [redux-orm/redux-orm#255]
First of all, thanks for awesome lib! I have an issue with List position change after bumping version from v0.12.4 to v0.13.0+. Selector doesn't get recomputed, but everything works properly when I add or remove an item. Also, all selectors which have some computation based on nested props have stopped working. Could I miss something?

List
```js
export default class extends Model {
  static modelName = 'List';

  static fields = {
    id: attr(),
    position: attr(),
    name: attr(),
    boardId: fk({
      to: 'Board',
      as: 'board',
      relatedName: 'lists',
    }),
  };

  ...
```
Reducer
```js
...

case ActionTypes.LIST_UPDATE_REQUESTED:
  List.withId(payload.id).update(payload.data);

  break;

...
```
Selector
```js
export const currentBoardListIdsSelector = createSelector(
  orm,
  dbSelector,
  state => pathSelector(state).boardId,
  ({ Board }, id) => {
    if (!id) {
      return null;
    }

    const boardModel = Board.withId(id);

    if (!boardModel) {
      return null;
    }

    return boardModel.lists
      .orderBy('position')
      .toRefArray()
      .map(list => list.id);
  },
);
```

**Comment by haveyaseen** [comment]
Hey @meltyshev, thanks for the kind words!

I haven't been able to find out what is not working exactly (see https://codesandbox.io/s/y03v08j7r9). Once I insert the relevant boards and/or items the selector is re-executed. Do you mind forking the sandbox and changing the code so that I can see what's going wrong?

**Comment by meltyshev** [comment]
Thank you for reply and sandbox! Built my situation here - https://codesandbox.io/s/m4xz3v9k8y. After List position changes the selector doesn’t hit and Lists order is the same as at first run. But if you change lib version back to 0.12.* everything is ok and selector runs twice.

**Comment by haveyaseen** [comment]
Awesome! This is definitely a bug, yes. The selector result is incorrectly being memoized due to an index-based access. Very valuable input, thanks a lot reporting this. I will be issuing a fix asap, for now please stick to v0.12.4 :slightly_smiling_face:

**Comment by haveyaseen** [comment]
Fixed and released as v0.13.3 (also fixed another bug, changelog will follow soon). Please check it out!
